homeassistant/components/sleepiq/binary_sensor.py

```diff
@@ -1,60 +1,55 @@
 """Support for SleepIQ sensors."""
-from __future__ import annotations
-
 from homeassistant.components.binary_sensor import (
     BinarySensorDeviceClass,
     BinarySensorEntity,
 )
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_USERNAME
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import SleepIQSensor
-from .const import DOMAIN, IS_IN_BED, SENSOR_TYPES, SIDES
+from . import DATA_SLEEPIQ, SleepIQDataUpdateCoordinator, SleepIQSensor
+from .const import BED, ICON_EMPTY, ICON_OCCUPIED, IS_IN_BED, SIDES
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the SleepIQ sensors."""
-    if discovery_info is None:
-        return
-
-    data = hass.data[DOMAIN]
-    data.update()
+    """Set up the SleepIQ bed binary sensors."""
+    coordinator = hass.data[DATA_SLEEPIQ].coordinators[config_entry.data[CONF_USERNAME]]
+    entities = []
 
-    dev = []
-    for bed_id, bed in data.beds.items():
+    for bed_id in coordinator.data:
         for side in SIDES:
-            if getattr(bed, side) is not None:
-                dev.append(IsInBedBinarySensor(data, bed_id, side))
-    add_entities(dev)
+            if getattr(coordinator.data[bed_id][BED], side) is not None:
+                entities.append(IsInBedBinarySensor(coordinator, bed_id, side))
+
+    async_add_entities(entities, True)
```

Comment:
```suggestion
    async_add_entities(entities)
```

These are coordinated, it shouldn't need to call for an update when adding